### ISSUE TITLE
Some OSes' hostname command doesn't have -f parameter

### ISSUE THREAD

**Issue by nanaya** [newrelic/rpm#306]
I got this error in my log:

```
usage: hostname [-s] [name-of-host]
hostname: unknown option -- f
```

And after some digging it seems to come from this:

https://github.com/newrelic/rpm/blob/418d9b854b4e65cddd9dbd10a4bc55cb4ceb5eb3/lib/new_relic/agent/hostname.rb#L21

For workaround I created a wrapper script which doesn't pass the `-f` parameter but it would be nice if there's some check if the system is Linux which only then pass that parameter as I believe other than in Linux it either isn't needed (defaults to full name in [FreeBSD](https://www.freebsd.org/cgi/man.cgi?query=hostname&apropos=0&sektion=0&manpath=FreeBSD+12.1-RELEASE+and+Ports&arch=default&format=html), [DragonFlyBSD](https://leaf.dragonflybsd.org/cgi/web-man?command=hostname&section=1), and [macOS](https://ss64.com/osx/hostname.html)) or just doesn't exist ([NetBSD](https://netbsd.gw.com/cgi-bin/man-cgi?hostname+1+NetBSD-6.0), [OpenBSD](https://man.openbsd.org/hostname.1), and [Solaris](https://docs.oracle.com/cd/E88353_01/html/E37839/hostname-1.html)).

**Comment by rachelleahklein** [comment]
Hi @nanaya! Thank you for bringing this to our attention. We are tracking this issue and plan to include a fix in an upcoming version of the agent.

**Comment by rachelleahklein** [comment]
The fix for this issue has been included in version 6.10.0 of the Ruby agent, released today! Thanks again for bringing it to our attention!
